File: backend/src/modules/profiles/application/use_cases/get_my_profile_use_case.py
from src.modules.profiles.domain.entities import Profile
from src.modules.profiles.domain.repositories import ProfileRepository
from src.core.errors.errors import NotFoundError


class GetMyProfileUseCase:
    """Caso de Uso para buscar os dados de perfil do usuário atual de forma BOLA-safe."""

    def __init__(self, repository: ProfileRepository):
        self.repository = repository

    async def execute(self, user_id: str) -> Profile:
        profile = await self.repository.get_by_id(user_id)
        if not profile:
            raise NotFoundError("Perfil não encontrado.")
        return profile

File: lawrence/lib/features/profile/application/use_cases/get_my_profile_use_case.dart
import '../../domain/entities/user_profile.dart';
import '../../domain/repositories/profile_repository.dart';

class GetMyProfileUseCase {
  const GetMyProfileUseCase(this._repository);

  final ProfileRepository _repository;

  Future<UserProfile> execute() {
    return _repository.getMyProfile();
  }
}
